multimap/src/main/java/org/infinispan/multimap/impl/SetBucket.java

```diff
@@ -53,6 +56,10 @@ Collection<MarshallableUserObject<V>> getWrappedValues() {
       return this.values.stream().map(MarshallableUserObject::new).collect(Collectors.toSet());
    }
 
+   public Set<V> values() {
+      return values;
```

Comment:
Not returning a copy might be causing an effect here. Using this in embedded, everything executes locally, which means that operations applied to the return also affect the bucket collection. I _suspect_ this could affect distributed mode if the requested node has the value locally.